src/v/cluster_link/tests/group_mirroring_task_test.cc

```diff
@@ -409,6 +427,47 @@ TEST_F(group_mirroring_task_test, test_happy_path_offsets_mirroring) {
       .get();
 }
 
+TEST_F(group_mirroring_task_test, test_find_coordinator_v3) {
+    make_current_node_leader_for({0, 1, 2, 3});
+    for (int i = 0; i < 3; ++i) {
+        ::model::node_id id{i};
+        fixture()->get_cluster_mock().set_supported_versions(
+          id,
+          kafka::find_coordinator_api::key,
+          {.min = kafka::find_coordinator_api::min_valid,
+           .max = kafka::api_version(3)});
+    }
+    fixture()->upsert_link(get_default_metadata()).get();
+
+    bool is_active = wait_for_task_state(model::task_state::active).get();
+    ASSERT_TRUE(is_active);
+
+    // set consumer group offsets
+
+    kafka::group_id test_group("t-group");
+
+    consumer_group_metadata metadata;
+    metadata.group_id = test_group;
+    metadata.coordinator_id = ::model::node_id(0);
+    source_cluster_group[test_group] = std::move(metadata);
```

Comment:
nitpick: Would it be more clear to use multiple groups & multiple coordinators? There might not be any benefit in terms of coverage.